### ISSUE TITLE
YLE problem or new API system?

### ISSUE THREAD

**Issue by gunnar12** [aajanki/yle-dl#240]
Maybe mid-day 15/6 monday "yle-dl stopped working"

'too many 503 error responses'))
for new "regular" programs, 
while the sports-50000 still "work"!.(but cant get publish-time!)

It seems yesterday and early morning "regulars" work like before, but  the ones published mid-day and later?

player.yle.fi gives 503, and probably some others too?

-- trace windoze ---

Now downloading from URL 2/108: https://areena.yle.fi/1-50337006
Traceback (most recent call last):
  File "C:\Users\xxxxx\AppData\Roaming\Python\Python37\site-packages\requests-2.19.1-py3.7.egg\requests\adapte
rs.py", line 445, in send
    timeout=timeout
  File "C:\Users\xxxxx\AppData\Roaming\Python\Python37\site-packages\urllib3-1.23-py3.7.egg\urllib3\connection
pool.py", line 731, in urlopen
    body_pos=body_pos, **response_kw)
  File "C:\Users\xxxxx\AppData\Roaming\Python\Python37\site-packages\urllib3-1.23-py3.7.egg\urllib3\connection
pool.py", line 731, in urlopen
    body_pos=body_pos, **response_kw)
  File "C:\Users\xxxxx\AppData\Roaming\Python\Python37\site-packages\urllib3-1.23-py3.7.egg\urllib3\connection
pool.py", line 731, in urlopen
    body_pos=body_pos, **response_kw)
  File "C:\Users\xxxxx\AppData\Roaming\Python\Python37\site-packages\urllib3-1.23-py3.7.egg\urllib3\connection
pool.py", line 711, in urlopen
    retries = retries.increment(method, url, response=response, _pool=self)
  File "C:\Users\xxxxx\AppData\Roaming\Python\Python37\site-packages\urllib3-1.23-py3.7.egg\urllib3\util\retry
.py", line 398, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='player.yle.fi', port=443): Max retries exceeded wit
h url: /api/v1/programs.jsonp?id=1-50337006&callback=yleEmbed.programJsonpCallback (Caused by ResponseError('to
o many 503 error responses'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "P:\ST31Xtra\audio\yle-dl-master\yle-dl-script.py", line 11, in <module>
    load_entry_point('yle-dl==20200419', 'console_scripts', 'yle-dl')()
  File "P:\ST31Xtra\audio\yle-dl-master\yledl\yledl.py", line 448, in main
    stream_filters, args.postprocess, args.metadatalang)
  File "P:\ST31Xtra\audio\yle-dl-master\yledl\yledl.py", line 301, in download
    title_formatter, io.ffprobe())
  File "P:\ST31Xtra\audio\yle-dl-master\yledl\extractors.py", line 350, in extract
    for clipurl in playlist]
  File "P:\ST31Xtra\audio\yle-dl-master\yledl\extractors.py", line 350, in <listcomp>
    for clipurl in playlist]
  File "P:\ST31Xtra\audio\yle-dl-master\yledl\extractors.py", line 578, in extract_clip
    pid, clip_url, title_formatter, ffprobe)
  File "P:\ST31Xtra\audio\yle-dl-master\yledl\extractors.py", line 801, in program_info_for_pid
    info = JSONP.load_jsonp(self.program_info_url(pid), self.httpclient)
  File "P:\ST31Xtra\audio\yle-dl-master\yledl\extractors.py", line 71, in load_jsonp
    httpclient.download_page(url, headers))
  File "P:\ST31Xtra\audio\yle-dl-master\yledl\http.py", line 45, in download_page
    response = self.get(url, extra_headers)
  File "P:\ST31Xtra\audio\yle-dl-master\yledl\http.py", line 81, in get
    r = self._session.get(url, headers=headers)
  File "C:\Users\xxxxx\AppData\Roaming\Python\Python37\site-packages\requests-2.19.1-py3.7.egg\requests\sessio
ns.py", line 525, in get
    return self.request('GET', url, **kwargs)
  File "C:\Users\xxxxx\AppData\Roaming\Python\Python37\site-packages\requests-2.19.1-py3.7.egg\requests\sessio
ns.py", line 512, in request
    resp = self.send(prep, **send_kwargs)
  File "C:\Users\xxxxx\AppData\Roaming\Python\Python37\site-packages\requests-2.19.1-py3.7.egg\requests\sessio
ns.py", line 622, in send
    r = adapter.send(request, **kwargs)
  File "C:\Users\xxxxx\AppData\Roaming\Python\Python37\site-packages\requests-2.19.1-py3.7.egg\requests\adapte
rs.py", line 504, in send
    raise RetryError(e, request=request)
requests.exceptions.RetryError: HTTPSConnectionPool(host='player.yle.fi', port=443): Max retries exceeded with
url: /api/v1/programs.jsonp?id=1-50337006&callback=yleEmbed.programJsonpCallback (Caused by ResponseError('too
many 503 error responses'))

**Comment by gunnar12** [comment]
WORST CASE POSSIBILITY!  YLE-DL KEYS BLACKLISTED??

Too fast, many connections per sec (10/s?). per hour (300/min??)  and "something"/ day??
Remembered something from API pages but cant find it again just now

Was forced to switch to new latest yle-dl , my old "own version" stopped working week ago.
Latest version different filename-formating, cant check if "already downloaded" for my already done files.

Decided solution, download ALL yle-audio-radio again (then compare same-crc files, delete duplicates with my own old crc-tool)

BUT!!

Noticed latest yle-dl "downloaded damxed FAST). wspwcially when "old series", some programs just 8-20MB.
My connection now 4G LTE, speed upto 50Mbps, 5MBps. Takes 3-6 seconds!!

My old "own version" used "speak" to speak "exists" and "N of M" inbetween downloads,  delay of 4-6 sec.. Clearly didnt break max access limit?

But "pure" yle-dl clearly exceeds max-limits,

- at least for json access to get title, check if already exists.
- probably also for "audio small files" when 4G or fiber speeds?

new YLE-KEY or reset needed, AND  pure delay, or some size-dependent delay to slow down yle-json, actual stream access?

PS  PS Just missed YLE Tek Help, had a "smart" guy last time.
PPS Was myself, although IP adress, blacklisted some 10-15 yrs ago due to "clearly automated access", Added that speak-delay but had probably left  "unproperly closed connections" too when file alrdy existed
(yle server closed only after its own time-out? might have had some 2-300 dangling connections worst case)

**Comment by gunnar12** [comment]
One more, 
wget seem to do some 503 lightning fast attempts?? lots of them 
 shld be changed, less or delay, timeouts

**Comment by aajanki** [comment]
It seems that the Areena API has changed. There is an initial fix on the master branch. This has not really anything to do with rate limitting. It's just a misleading error message from the Areena server.

I'll test a bit more and close this issue later if everything seem to be working.

**Comment by ernie77** [comment]
Here is current API

https://developer.yle.fi/en/api/index.html

I tried this new api address, with my YLE tunnus and found small change in the json.

https://external.api.yle.fi/v1/programs/items/1-200102.json?app_key=YOUR_APP_KEY&app_id=YOUR_APP_ID

**Comment by gunnar12** [comment]
Thanks!

But no subtext files for mp4 format 

I have tested only audio and ffmpeg preferred mp4.

**Comment by gunnar12** [comment]
simple grep, still one place where old API=url used:


    def yle_media_descriptor(self, program_id, media_id, protocol):
        media_jsonp_url = 'https://player.yle.fi/api/v1/media.jsonp?' \
                          'id=%s&callback=yleEmbed.startPlayerCallback&' \
                          'mediaId=%s&protocol=%s&client=areena-flash-player' \
                          '&instance=1' % \
            (quote_plus(media_id), quote_plus(program_id),
             quote_plus(protocol))
        media = JSONP.load_jsonp(media_jsonp_url, self.httpclient)

        if media:
            logger.debug('media:\n' + json.dumps(media, indent=2))

        return media

**Comment by aajanki** [comment]
I released version 20200618 which fixes the recent problems with Areena.
